de/nmichael/efa/data/storage/Journal.java

```diff
@@ -316,12 +316,14 @@ public boolean log(long scn, Operation operation, DataRecord r) {
             if (s == null || s.length() == 0) {
                 Logger.log(Logger.ERROR, Logger.MSG_DATA_JOURNALWRITEFAILED,
                         LogString.fileWritingFailed(fwname, International.getString("Journal"), "empty log string"));
+                f.close(); // close the writer to avoid resource leak
```

Comment:
@cubitusgh2 No, we can't close f here. openForAppend() does not create a new BufferedWriter every time. We're reusing the same BufferedWriter for every log write until we rotate the journal. If we were to close it here, subsequent writes to the journal would fail.